client/util/calculations.test.js

```diff
@@ -0,0 +1,175 @@
+import calculations from './calculations'
+
+const purchasePrice = 500000
+let interestRate = 0.045
+let downPayment = 0.2
+let closingCosts = 0.03
+let salesCommission = 0.06
+let propertyTaxRate = 0.0125
+const hoa = 400
+const maintenance = 100
+const insurance = 100
+let annualAppreciationRate = 0.03
+let incomeTaxRate = 0.25
+let generalInflationRate = 0.02
+const rentBase = 2500
+let rentInflationRate = 0.02
+let rentReturn = 0.06
+const principal = 400000
+
+const homeValue = calculations.forecastHomeValue(purchasePrice, annualAppreciationRate)
+const mortgagePayment = calculations.calculateMortgagePayment(principal, interestRate)
+const { debt, paidInterest, paidPrincipal } = calculations.forecastDebt(principal, interestRate, mortgagePayment)
+const equity = calculations.forecastEquity(homeValue, debt)
+const fees = calculations.forecastSellingFees(homeValue, salesCommission)
+const netEquity = calculations.forecastNetEquity(equity, fees)
+const propertyTax = calculations.forecastPropertyTax(homeValue, propertyTaxRate)
+const expenses = calculations.forecastExpenses(propertyTax, hoa, maintenance, insurance, generalInflationRate)
+const deductions = calculations.forecastDeductions(paidInterest, propertyTax, incomeTaxRate)
+const cashFlow = calculations.forecastCashFlow(deductions, mortgagePayment, expenses)
+const rent = calculations.forecastRent(rentBase, rentInflationRate)
+const savings = calculations.forecastSavings(cashFlow, rent)
+const investment = calculations.forecastRentInvestment(downPayment, closingCosts, purchasePrice, rentReturn, savings)
+
+describe('Home value', () => {
+  it('Calculates the home value correctly.', () => {
```

Comment:
Declare `purchasePrice` and calculate `homeValue` within the test (`it`). Keep your unit tests isolated from each other and no state between them.